### PR TITLE
Bump deps

### PR DESCRIPTION
**Pull request by kevmoo**

### CODE REVIEW

File: packages/flutter_image/example/pubspec.yaml [hunk 1]
```diff
@@ -17,7 +17,7 @@ dependencies:
     path: "../"
 
 dev_dependencies:
-  flutter_lints: ^2.0.0
+  flutter_lints: ^4.0.0
```
Comment (stuartmorgan) [inline packages/flutter_image/example/pubspec.yaml:20]: This dependency is just a mistake, and should be removed. It's in the template and sometimes we forget to remove it.

File: packages/two_dimensional_scrollables/example/pubspec.yaml [hunk 2]
```diff
@@ -16,7 +16,7 @@ dependencies:
     path: ../
 
 dev_dependencies:
-  flutter_lints: ^2.0.0
+  flutter_lints: ^4.0.0
```
Comment (stuartmorgan) [inline packages/two_dimensional_scrollables/example/pubspec.yaml:19]: Same.

File: packages/two_dimensional_scrollables/pubspec.yaml [hunk 3]
```diff
@@ -13,7 +13,7 @@ dependencies:
     sdk: flutter
 
 dev_dependencies:
-  flutter_lints: ^2.0.0
+  flutter_lints: ^4.0.0
```
Comment (stuartmorgan) [inline packages/two_dimensional_scrollables/pubspec.yaml:16]: Same.

File: packages/flutter_image/example/pubspec.yaml (hunk 1 above)
Comment (kevmoo) [reply packages/flutter_image/example/pubspec.yaml:20]: 👍

**Review by goderbauer — COMMENTED** [review]
LGTM for changes to `packages/flutter_lints`

**Review by Piinks — APPROVED** [review]
LGTM for two_dimensional_scrollables

**Review by stuartmorgan — APPROVED** [review]
LGTM

**Pull request merged**